code/parachain/frame/pablo/src/lib.rs

```diff
@@ -847,6 +849,8 @@ pub mod pallet {
 			min_receive: AssetAmount<Self::AssetId, Self::Balance>,
 			keep_alive: bool,
 		) -> Result<SwapResult<Self::AssetId, Self::Balance>, DispatchError> {
+			ensure!(in_asset.asset_id != min_receive.asset_id, Error::<T>::CannotSwapSameAsset);
```

Comment:
`do_buy` may need the same validation

Reply:
done: https://github.com/ComposableFi/composable/commit/747b32b8934a40c0f67b59f5ec4edbde4b9e1951